crates/esplora/src/async_ext.rs

```diff
@@ -6,11 +6,17 @@ use bdk_chain::{
     local_chain::{self, CheckPoint},
     BlockId, ConfirmationTimeAnchor, TxGraph,
 };
-use esplora_client::{Error, TxStatus};
+use esplora_client::TxStatus;
 use futures::{stream::FuturesOrdered, TryStreamExt};
 
 use crate::{anchor_from_status, ASSUME_FINAL_DEPTH};
 
+/// `Box` a large [`esplora_client::Error`]
+type Error = Box<esplora_client::Error>;
+
+/// Alias `Result` to use our `Error`
+type Result<T, E = Error> = core::result::Result<T, E>;
```

Comment:
Why is this line needed?

Reply:
You're right it's not needed, let's get rid of it. The thought was that it allows omitting the `Err` variant in a function signature similar to what you get from `anyhow::Result`, but I neglected to actually remove them. In this case, it's probably better to be explicit and keep `Error` in the fn signatures.